misc/pytorch_toolkit/text_recognition/text_recognition/data/utils.py

```diff
@@ -441,21 +441,26 @@ def get_num_lines_in_file(path):
 
 
 def collate_fn(sign2id, batch, *, batch_transform=None, use_ctc=False):
-    # filter the pictures that have different width or height
-    size = batch[0]['img'].shape
-    batch = [img_text for img_text in batch if img_text['img'].shape == size]
     # sort by the length of text
     # the purpose of the sort is to put the longest text on the first place
     # to get correct size of the tensor in the texts2tensor function
     batch.sort(key=lambda img_text: len(img_text['text'].split()), reverse=True)
 
+    imgs = [item['img'] for item in batch]
+    if batch_transform:
+        imgs = batch_transform(imgs)
+    for i, item in enumerate(batch):
```

Comment:
is it kind of redundant activity if batch_transform is False?

Reply:
together with line 449

Reply:
Moved this code under if condition.